src/zenml/zen_server/utils.py

```diff
@@ -177,6 +183,18 @@ def handle_exceptions(func: F) -> F:
 
     @wraps(func)
     def decorated(*args: Any, **kwargs: Any) -> Any:
+        from zenml.zen_server.auth import AuthContext, set_auth_context
```

Comment:
@stefannica @fa9r Can you explain what changed with the new analytics that this decorator now needs to to this?

Reply:
@schustmi the new analytics need to access the "active user" to use the active user ID as the canonical event ID. When running server-side, every REST API request has a different "active user". One way to fetch it in the analytics is to pass it down the call stack as a function/method argument, from the API routers all the way to the analytics utils, but this is laborious, as we need to change the signature of many functions/methods to make this happen.

This approach is cleaner:
* we hack this decorator that is already applied to all routers
* here we fetch the active user from the authentication context and save it in a "ContextVar", which is a thread-safe global variable
* we load the user from the global variable in the analytics utils code

Reply:
Ah that makes a lot of sense, thank you very much!